tests/jballerina-unit-test/src/test/resources/test-src/services/client_resource_declaration_negative.bal

```diff
@@ -0,0 +1,78 @@
+string? stringNilArgument = ();
+
+public function testClientDeclarationError() {
+    var errorClient = client object {
+        resource function get path/[int b]/foo(string name2) returns int {
+            return 1;
+        }
+
+        resource function get path/[int a]/foo/[string a](string name) returns int {
+            return 1;
+        }
+
+        resource function get path/[int a]/foo2(string name, string name) returns int {
+            return 1;
+        }
+
+        resource function get path/[int a]/foo2(string name, string address) {
+
+        }
+
+        resource function get path/[int... a]/[int... a]() returns int {
+            return 1;
+        }
+
+        resource function post path/[string? c]/foo(string name2) returns string? {
+            return stringNilArgument;
+        }
+
+        resource function path() returns int {
+            return 1;
+        }
+
+        resource function get recordPath/[record {int a;}]() returns int {
+            return 1;
+        }
+
+        resource function get recordPath2/[record {int a;} a]() returns int {
+            return 1;
+        }
+
+        resource function get xmlPath/[xml... a]() returns int {
+            return 1;
+        }
+
+        resource function get xmlPath2/[xml a]() returns int {
+            return 1;
+        }
+
+        resource function get xmlPath2/[xml]() returns int {
+            return 1;
+        }
+
+        resource function get mapPath/[map<string>]() returns int {
+            return 1;
+        }
+
+        resource function get mapPath2/[map<string> a]() returns int {
+            return 1;
+        }
+
+        resource function get intOrErrorPath/[int|error]() returns int {
+            return 1;
+        }
+
+        resource function get intOrErrorPath2/[int|error a]() returns int {
+            return 1;
+        }
+
+        resource function get errorPath/[error]() returns int {
+            return 1;
+        }
+
+        resource function get errorPath2/[error a]() returns int {
+            return 1;
+        }
+
+    };
+}
```

Comment:
@SasinduDilshara, Shall we add few tests with valid/invalid module-type-defn?

Reply:
Also we need to a tests for invalid returns

Reply:
We have invalid return tests in `client_resource_params_negative.bal`.
Do we need add more?

Reply:
I will add tests for `valid/invalid module-type-defn`

Reply:
Added. b731811e1ec16fc62289fbdaff5f454fd9dd283c